arch/cortex-m0/src/lib.rs

```diff
@@ -77,7 +73,7 @@ pub unsafe extern "C" fn switch_to_user(mut user_stack: *const u8,
 
     "
     : "=r"(user_stack)
-    : "r"(user_stack), "r"(process_got), "r"(process_regs)
+    : "r"(user_stack), "r"(process_regs)
```

Comment:
I'm not sure I agree, and it does impact the assembly. I'll fix everything else then come back to this. We may need to convince ourselves a bit.